core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -160,11 +161,14 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
 
   override def preStart(): Unit = {
     super.preStart()
+
     consumer = settings.createKafkaConsumer()
+    currentPollTask = schedulePollTask()
   }
 
   override def postStop(): Unit = {
-    pollTask.cancel()
+    if (currentPollTask != null)
+      currentPollTask.cancel()
```

Comment:
Ah, wait, that cannot work - `this` isn't an ActorRef. Is there a way to implement this when comparing `self`?